badge-maker/lib/index.js

```diff
@@ -42,23 +52,26 @@ function _clean(format) {
 
   const cleaned = {}
   Object.keys(format).forEach(key => {
-    if (format[key] != null && expectedKeys.includes(key)) {
-      cleaned[key] = format[key]
-    } else {
-      throw new ValidationError(
-        `Unexpected field '${key}'. Allowed values are (${expectedKeys.toString()})`
-      )
+    if (format[key] != null) {
+      if (expectedKeys.includes(key)) {
+        cleaned[key] = format[key]
+      } else {
+        throw new ValidationError(
+          `Unexpected field '${key}'. Allowed values are (${expectedKeys.toString()})`
+        )
+      }
     }
   })
 
-  // convert "public" format to "internal" format
-  cleaned.text = [cleaned.label || '', cleaned.message]
-  delete cleaned.label
-  delete cleaned.message
-  if ('style' in cleaned) {
-    cleaned.template = cleaned.style
-    delete cleaned.style
-  }
+  // Whitespace removal.
+  cleaned.label = `${cleaned.label}`.trim()
```

Comment:
For a message only badge, running
```
cleaned.label = `${cleaned.label}`.trim()
```

would result in the label property being the string literal `"undefined"`. This is not what we want.

Reply:
Indeed. I'll fix this up tomorrow.